### PR TITLE
#290: Enhanced javadoc for the Response class

### PR DESCRIPTION
**Pull request by bdragan**
#290: Enhanced javadoc for the Response class by describing the restrictions regarding the order of the head() and body() methods invocation.

### CODE REVIEW

**Comment by davvd** [comment]
@bdragan Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@pinaf this pull request is for you,please review

File: src/main/java/org/takes/Response.java [hunk 1]
```diff
@@ -43,6 +43,11 @@
  *   "Content-Type", "text/html"
  * );</pre>
  *
+ * <p>The implementations of this interface may require that
+ * {@link Response#head()} method has to be invoked before reading from the
+ * {@code InputStream}, but they must NOT require that the {@link InputStream}
```
Comment (pinaf) [inline src/main/java/org/takes/Response.java]: @bdragan by `{@code InputStream}`, do you mean `Response#body()`?

File: src/main/java/org/takes/Response.java [hunk 2]
```diff
@@ -43,6 +43,11 @@
  *   "Content-Type", "text/html"
  * );</pre>
  *
+ * <p>The implementations of this interface may require that
```
Comment (pinaf) [inline src/main/java/org/takes/Response.java]: @bdragan are there any implementations with this requirement? If so, we should document them accordingly.

**Comment by pinaf** [comment]
@bdragan please see 2 comments above.

**Comment by bdragan** [comment]
@pinaf Thanks, please review the new commit.

> by `{@code InputStream}`, do you mean `Response#body()`?

Yes, I updated the javadoc to be more more precise about this.

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, check this issue for QA compliance, as per [par.24](http://www.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@pinaf 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were only two issues found during review.
Please confirm that you will try to find at least three problems next time.

**Comment by pinaf** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
@pinaf just added 10 mins to @elenavolokhova (for QA), payment ID is `67988941`; thank you, added **20 mins** to your acc, payment num is `AP-23C77134L33168444`, 61 hours and 54 mins spent in total to complete; review comments (c=5) added as a bonus; added +20 to your rating, now it is equal to [+4258](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**